util/host-utils.c

```diff
@@ -266,3 +266,154 @@ void ulshift(uint64_t *plow, uint64_t *phigh, int32_t shift, bool *overflow)
         *plow = *plow << shift;
     }
 }
+/*
+ * Unsigned 256-by-128 division.
+ * Returns the remainder via r.
+ * Returns lower 128 bit of quotient.
+ */
+static Int128 udiv256_qrnnd(Int128 *r, Int128 n0, Int128 n1, Int128 d)
+{
+    Int128 d0, d1, q0, q1, r1, r0, m;
+    uint64_t mp0, mp1;
+
+    d0 = int128_make64(int128_getlo(d));
+    d1 = int128_make64(int128_gethi(d));
+
+    r1 = int128_remu(n1, d1);
+    q1 = int128_divu(n1, d1);
+    mulu64(&mp0, &mp1, int128_getlo(q1), int128_getlo(d0));
+    m = int128_make128(mp0, mp1);
+    r1 = int128_or(int128_lshift(r1, 64), int128_make64(int128_gethi(n0)));
```

Comment:
I believe this is equivalent to ``int128_make128(int128_gethi(n0), int128_getlo(r1)``